### PR TITLE
Feature/1317 block content three column

### PR DESCRIPTION
**Pull request by sagarsdeshmukh**

### CODE REVIEW

File: classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+			if ( is_array( $image_2_data_array ) && $image_2_data_array ) {
+				$images_data['image_2_src']   = $image_2_data_array[0];
+				$images_data['alt_2_text']    = get_post_meta( $fields['image_2'], '_wp_attachment_image_alt', true  );
+			}
+
+			if ( is_array( $image_3_data_array ) && $image_3_data_array ) {
+				$images_data['image_3_src']   = $image_3_data_array[0];
+				$images_data['alt_3_text']    = get_post_meta( $fields['image_1'], '_wp_attachment_image_alt', true  );
+			}
```
Comment (tcrsavage) [inline classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php]: 93-112 could go in a `for( $i=1; $i<3; $i++)` loop.

File: classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+			if ( is_array( $image_2_data_array ) && $image_2_data_array ) {
+				$images_data['image_2_src']   = $image_2_data_array[0];
+				$images_data['alt_2_text']    = get_post_meta( $fields['image_2'], '_wp_attachment_image_alt', true  );
+			}
+
+			if ( is_array( $image_3_data_array ) && $image_3_data_array ) {
+				$images_data['image_3_src']   = $image_3_data_array[0];
+				$images_data['alt_3_text']    = get_post_meta( $fields['image_1'], '_wp_attachment_image_alt', true  );
```
Comment (tcrsavage) [inline classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php]: Looks like this wants to be `$fields['image_3']`

File: classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		 *
+		 * @since 0.1.0
+		 *
+		 * @return string All the data used for the html.
+		 */
+		public function prepare_template( $fields, $content, $shortcode_tag ) : string {
+
+			for ( $i = 1; $i < 4; $i++ ) {
+				$img_array = wp_get_attachment_image_src( $fields[ "image_$i" ], 'full' );
+				if ( false !== $img_array && ! empty( $img_array ) ) {
```
Comment (angtheod) [inline classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php]: @sagarsdeshmukh this if condition can be simplified to `if ( $img_array )`

File: classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+		/**
+		 * Shortcode UI setup for the ThreeColumn shortcode.
+		 *
+		 * It is called when the Shortcake action hook `register_shortcode_ui` is called.
+		 *
+		 * @since 0.1.0
+		 */
+		public function prepare_fields() {
+			$fields = array(
```
Comment (angtheod) [inline classes/controller/blocks/class-p4bks-blocks-contentthreecolumn-controller.php]: @sagarsdeshmukh change those to short syntax for arrays []

**Pull request merged**